pkg/types/resource/user_trait.go

```diff
@@ -34,6 +34,22 @@ func WithEmail(email string, primary bool) UserTraitOption {
 		return nil
 	}
 }
+func WithUsername(username string, primary bool) UserTraitOption {
```

Comment:
I'm not sure `Username` is the right name for this field. If it is, this option should probably be renamed to `WithUserUsername()` since it is an option specific to user traits. 

maybe `external_identifier` ?